api/v1/inferencepool_types.go

```diff
@@ -61,19 +61,28 @@ type InferencePoolSpec struct {
 	// +kubebuilder:validation:Required
 	Selector map[LabelKey]LabelValue `json:"selector"`
 
-	// TargetPortNumber defines the port number to access the selected model server Pods.
-	// The number must be in the range 1 to 65535.
+	// TargetPorts defines the ports to access the selected model server Pods.
 	//
-	// +kubebuilder:validation:Minimum=1
-	// +kubebuilder:validation:Maximum=65535
 	// +kubebuilder:validation:Required
-	TargetPortNumber int32 `json:"targetPortNumber"`
+	// +kubebuilder:validation:MinItems=1
+	// +kubebuilder:validation:MaxItems=1
+	TargetPorts []Port `json:"targetPorts"`
```

Comment:
```suggestion
    // +listType=Set
    // +listTypeKey=number
	TargetPorts []Port `json:"targetPorts"`
```

I'm not quite sure if these are the right tags, but want something to ensure that this list works well.

Reply:
done

Reply:
As port is not a built in type as string.


I need to use 
```	
// +listType=map
// +listMapKey=number
```